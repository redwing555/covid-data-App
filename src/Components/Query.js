import React from 'react';
import Proptypes from 'prop-types';
import '../styles/query.css';

const Query = ({ handleChange }) => (
  <form className="form">
    <span className="search-by">Stats by Country</span>
    <input className="input-search" type="text" placeholder="Enter country name" onChange={handleChange} />
  </form>
);

Query.propTypes = {
  handleChange: Proptypes.func.isRequired,
};

export default Query;
