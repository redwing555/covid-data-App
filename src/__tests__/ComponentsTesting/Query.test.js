import React from 'react';
import renderer from 'react-test-renderer';
import Query from '../../Components/Query';

describe('Searching', () => {
  it('renderers search correctly', () => {
    const tree = renderer
      .create(<Query handleChange={() => 7 + 4} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
