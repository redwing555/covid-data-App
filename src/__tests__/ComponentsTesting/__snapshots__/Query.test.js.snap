// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searching renderers search correctly 1`] = `
<form
  className="form"
>
  <span
    className="search-by"
  >
    Stats by Country
  </span>
  <input
    className="input-search"
    onChange={[Function]}
    placeholder="Enter country name"
    type="text"
  />
</form>
`;
